server/src/controllers/taskController.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+const createTask = require('../services/createTask');
+const updateTask = require('../services/updateTask');
+const deleteTask = require('../services/deleteTask');
+const markTaskAsCompleted = require('../services/markTaskAsCompleted');
+const markTaskAsUncompleted = require('../services/markTaskAsUncompleted');
+const getAllCompletedTasks = require('../services/getAllCompletedTasks');
+const getAllUncompletedTasks = require('../services/getAllUncompletedTasks');
+const getAllUserTasks = require('../services/getAllUserTasks');
+const getAllOverdueTasks = require('../services/getAllOverdueTasks');
+
+class TaskController {
+  async createTask(req, res) {
+    try {
+      const userID = req.userID;
+
+      const task = await createTask({userID, ...req.body });
+
+      res.status(201).json({ 
+        status: 'success', 
+        task 
+      });
+    } catch (err) {
+      res.status(400).json({ 
+        status: 'failed', 
+        message: err.message 
+      });
+    }
+  }
+
+  async updateTask(req, res) {
+    try {
+      const userID = req.userID;
+      const taskID = req.body.id;
+
+      delete req.body.id;
+
+      const task = await updateTask(userID, taskID, req.body);
+
+      res.status(200).json({ 
+        status: 'success', 
+        task
+      });
+    } catch (err) {
+      res.status(400).json({ 
+        status: 'failed', 
+        message: err.message 
+      });
+    }
+  }
+
+  async deleteTask(req, res) {
+    try {
+      const userID = req.userID;
+      const taskID = req.body.id;
+
+      const deletedTask = await deleteTask(userID, taskID);
+
+      res.status(200).json({ 
+        status: 'success', 
+        deletedTask
+      });
+    } catch (err) {
+      res.status(400).json({ 
+        status: 'failed', 
+        message: err.message  
+      });
+    }
+  }
+
+  async markTaskAsCompleted(req, res) {
```

Comment:
You can create only one method to mark task as done/undone instead of ```markTaskAsUncompleted``` and ```markTaskAsCompleted```, just name it ```toggleTask``` and just manipulate with isDone value based on previous value of it. Just my own opinion 😅